external/fd/pddl/conditions.py

```diff
@@ -277,7 +277,10 @@ class Literal(Condition):
     parts = []
     def __init__(self, predicate, args):
         self.predicate = predicate
-        self.args = tuple(args)
+        if any(isinstance(x, list) for x in args):
+            self.args = tuple(tuple(x) for x in args)
+        else:
+            self.args = tuple(args)
```

Comment:
Hi @noskill,

how does this relate to the double free issue? Also, what is the rationale for the change above? When can the `args` parameter contain instances of `list`?